### PR TITLE
11.0 mig account_invoice_start_end_dates

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: account_invoice_start_end_dates/README.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* the module *sale_start_end_dates* from the sale-workflow OCA project: this module adds the fields *Start Date* and *End Date* on sale order lines and copies the information from sale order lines to invoice lines.
+
+* the module *account_cutoff_prepaid* (same repository): this module allows easy computation of prepaid expenses and prepaid revenues.
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/89/10.0
```
Comment (simahawk) [inline account_invoice_start_end_dates/README.rst]: /11.0

File: account_invoice_start_end_dates/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# © 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
```
Comment (simahawk) [inline account_invoice_start_end_dates/__manifest__.py]: replace '(c)' w/ 'Copyright'

File: account_invoice_start_end_dates/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Account Invoice Start End Dates',
+    'version': '11.0.1.0.0',
+    'category': 'Accounting & Finance',
+    'license': 'AGPL-3',
+    'summary': 'Adds start/end dates on invoice lines and move lines',
+    'author': 'Akretion,Odoo Community Association (OCA)',
+    'website': 'http://www.akretion.com',
```
Comment (simahawk) [inline account_invoice_start_end_dates/__manifest__.py]: https://github.com/OCA/account-closing

File: account_invoice_start_end_dates/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    def inv_line_characteristic_hashcode(self, invoice_line):
+        """Add start and end dates to hashcode used when the option "Group
+        Invoice Lines" is active on the Account Journal"""
+        code = super().inv_line_characteristic_hashcode(
+            invoice_line)
+        hashcode = '%s-%s-%s' % (
+            code,
+            invoice_line.get('start_date', 'False'),
+            invoice_line.get('end_date', 'False'),
+            )
```
Comment (simahawk) [inline account_invoice_start_end_dates/models/account_invoice.py]: indent back

**Review by simahawk — CHANGES_REQUESTED** [review]
travis broken

**Comment by fclementic2c** [comment]
Functional test Ok
:+1:

**Comment by fclementic2c** [comment]
...however this module is useless without : https://github.com/OCA/account-closing/pull/60

File: account_invoice_start_end_dates/models/account_invoice.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    def inv_line_characteristic_hashcode(self, invoice_line):
+        """Add start and end dates to hashcode used when the option "Group
+        Invoice Lines" is active on the Account Journal"""
+        code = super().inv_line_characteristic_hashcode(
+            invoice_line)
```
Comment (mpanarin) [inline account_invoice_start_end_dates/models/account_invoice.py]: newline the parenthesis

**Review by mpanarin — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by mpanarin — APPROVED** [review]
LGTM + tested

**Review by tschanzt — APPROVED** [review]
looks good to me.

File: account_invoice_start_end_dates/__manifest__.py (hunk 2 above)
Comment (simahawk) [reply account_invoice_start_end_dates/__manifest__.py]: @oleksandrpaziuk ping

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
reminder: migration commits squash needed before merge

**Comment by fclementic2c** [comment]
LGTM
:+1:

**Pull request merged**